class-bookrestcontroller.php

```diff
@@ -0,0 +1,311 @@
+							'type'              => 'string',
+							'sanitize_callback' => 'sanitize_text_field',
+							'validate_callback' => array( $this, 'validate_request_param_text' ),
+						),
+					),
+				),
+			)
+		);
+		//patch for edit, not all fields must be sent
+		register_rest_route(
+			$this->namespace,
+			'/book/(?P<id>\d+)',
+			array(
+				'methods'             => WP_REST_Server::EDITABLE,
+				'callback'            => array( $this, 'update_item' ),
+				'permission_callback' => array( $this, 'get_user_can_edit' ),
+				'args'                => array(
+					'id'      => array(
+						'required'          => true,
+						'validate_callback' => array( $this, 'validate_int' ),
+					),
+					'post_id' => array(
+						'required'          => true,
+						'validate_callback' => array( $this, 'validate_int' ),
+					),
+					'title'   => array(
+						'required'          => false,
+						'type'              => 'string',
+						'sanitize_callback' => 'sanitize_text_field',
+						'validate_callback' => array( $this, 'validate_request_param_name' ),
+					),
+					'author'  => array(
+						'required'          => false,
+						'type'              => 'string',
+						'sanitize_callback' => 'sanitize_text_field',
+						'validate_callback' => array( $this, 'validate_request_param_name' ),
+					),
+					'genre'   => array(
+						'required'          => false,
+						'type'              => 'string',
+						'sanitize_callback' => 'sanitize_text_field',
+						'validate_callback' => array( $this, 'validate_request_param_name' ),
+					),
+					'summary' => array(
+						'required'          => false,
+						'type'              => 'string',
+						'sanitize_callback' => 'sanitize_text_field',
+						'validate_callback' => array( $this, 'validate_request_param_text' ),
+					),
+				),
+			)
+		);
+		//delete item
+		register_rest_route(
+			$this->namespace,
+			//use id and post_id as request parameters since both are needed in the current format
+			'/book/(?P<id>\d+)/(?P<post_id>\d+)',
+			array(
+				'methods'             => WP_REST_Server::DELETABLE,
+				'callback'            => array( $this, 'delete_item' ),
+				'permission_callback' => array( $this, 'get_user_can_edit' ),
+				'args'                => array(
+					'id' => array(
+						'required'          => true,
+						'validate_callback' => array( $this, 'validate_int' ),
+					),
+				),
+				'header'              => 'post_id : post_id',
+			)
+		);
+	}
+
+	/**
+	 * Retrieves all items from the post meta in JSON format<br>
+	 *
+	 * @param WP_REST_Request $request MUST have the key 'id'
+	 *
+	 * @return WP_REST_Response The data in format JSON or FALSE if bad request
+	 */
+	public function get_items( $request ) {
+		$response = new WP_REST_Response();
+		if ( ! $this->validate_int( $request->get_param( 'id' ) ) ) {
+			$response->set_status( 400 );
+			return $response;
+		}
+
+		$data   = get_post_meta( (int) $request->get_param( 'id' ), self::META_KEY, true );
```

Comment:
We could use the `BookService::get_items( (int) $request->get_param( 'id' ) );` here